File: 0061-rotate-list/0061-rotate-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} k
 * @return {ListNode}
 */
var rotateRight = function(head, k) {
    
    let array = new Array(500);
    let temp = head;
    let len = 0;

    while(temp != null)
    {
        array[len++] = temp.val;
        temp = temp.next;
    }

    temp = head;
    k = k % len;
    
    if(k === 0)
         return head;

    for(let i = 0 ; i < k ; i++)
    {
        temp.val = array[len - k + i];
        temp = temp.next;
    }
    
    for(let i = 0 ; i < len - k ; i++)
    {
        temp.val = array[i];
        temp = temp.next;
    }

    return head;
};

File: 0061-rotate-list/0061-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        
        if(head == null || head.next == null)
            return head;

        ListNode tail = head;
        int len = 1;
        
        while(tail.next != null)
        {
            tail = tail.next;
            len++;
        }

        if(k % len == 0)
            return head;

        tail.next = head;
        ListNode newLastNode = findNthNode(head, len - (k % len));

        head = newLastNode.next;
        newLastNode.next = null;
    
        return head;
    }

    public ListNode findNthNode(ListNode head, int n)
    {

        ListNode temp = head;
        int cnt = 1;

        while( temp != null)
        {
            if(cnt == n)
                return temp;

            cnt++;
            temp = temp.next;
        }

        return temp;
    }
}